Sessions migration guide (#783)
Qiskit/documentation: docs/run/run-jobs-in-session.mdx

There are several ways to set up and use [sessions](sessions).  It is recommended that you do not run a session with a single job in it.

<Admonition type="note">

> Session execution mode is not supported in the Open plan. Jobs will run in job mode instead.

Reviewer: ```suggestion
Session execution mode is not supported in the Open Plan. Jobs will run in job mode instead.
```

<span id="specify-length"></span>
## Session length

You can define the maximum session timeout with the `max_time` parameter. This should exceed the longest job's execution time and be within the system's limits.

``` python
with Session(service=service, backend=backend, max_time="25m"):
  ...
```

There is also an interactive timeout value (ITTL, or interactive time to live) that cannot be configured.  If no session jobs are queued within that window, the session is temporarily deactivated. To determine a session's ITTL, follow the instructions in [Determine session details](#session-details) and look for the `interactive_timeout` value.

<span id="close"></span>
## Close a session

> A session automatically closes when it exits the context manager. With `qiskit-ibm-runtime` 0.13 or later releases, when the session context manager is exited, the session is put into "In progress, not accepting new jobs" status. This means that the session finishes processing all running or queued jobs until the maximum timeout value is reached. After all jobs are completed, the session is immediately closed. This allows the scheduler to run the next job without waiting for the session interactive timeout, thereby reducing the average job queuing time. You cannot submit jobs to a closed session.

Reviewer 2: Should this be cancel since the session is already closed and cancel matches the qiskit call equivalent to what is happening
```suggestion
A session automatically closes when it exits the context manager. With `qiskit-ibm-runtime` 0.13 or later releases, when the session context manager is exited, the session is put into "In progress, not accepting new jobs" status. This means that the session finishes processing all running or queued jobs until the maximum timeout value is reached. After all jobs are completed, the session is immediately canceled. This allows the scheduler to run the next job without waiting for the session interactive timeout, thereby reducing the average job queuing time. You cannot submit jobs to a closed session.
```

<span id="close"></span>
## Close a session

A session automatically closes when it exits the context manager. With `qiskit-ibm-runtime` 0.13 or later releases, when the session context manager is exited, the session is put into "In progress, not accepting new jobs" status. This means that the session finishes processing all running or queued jobs until the maximum timeout value is reached. After all jobs are completed, the session is immediately canceled. This allows the scheduler to run the next job without waiting for the session interactive timeout, thereby reducing the average job queuing time. You cannot submit jobs to a closed session.

``` python
with Session(service=service, backend=backend) as session:
    estimator = Estimator()
    job1 = estimator.run(...)
    job2 = estimator.run(...)

# The session is no longer accepting jobs but the submitted job will run to completion.
result = job1.result()
result2 = job2.result()
```

> If you are not using a context manager, manually close the session after all the necessary results have been retrieved. With `qiskit-ibm-runtime` 0.13 or later releases, when a session is closed with `session.close()`, it no longer accepts new jobs, but the already submitted jobs will still run until completion and their results can be retrieved. Prior to qiskit-ibm-runtime 0.13, when a session is closed with `session.close()`, any jobs that are already running continue to run, but any queued jobs remaining in the session are put into a failed state.

Reviewer 3: ```suggestion
If you are not using a context manager, you should manually close the session to avoid unwanted cost. With `qiskit-ibm-runtime` 0.13 or later releases, you can close a session as soon as you are done submitting jobs to it. When a session is closed with `session.close()`, it no longer accepts new jobs, but the already submitted jobs will still run until completion and their results can be retrieved. Prior to qiskit-ibm-runtime 0.13, when a session is closed with `session.close()`, any jobs that are already running continue to run, but any queued jobs remaining in the session are put into a failed state.
```

session.close()
```

<span id="cancel"></span>
## Cancel a session

Canceling a session immediately closes it, failing all queued jobs and preventing new submission. Use the `session.cancel()` method to cancel a session.  Any jobs that are already running continue to run but queued jobs are put into a failed state and no further jobs can be submitted to the session. This is a convenient way to quickly fail all queued jobs within a session.

``` python
with Session(service=service, backend=backend) as session:
    estimator = Estimator()
    job1 = estimator.run(...)
    job2 = estimator.run(...)
    # You can use session.cancel() to fail all pending jobs, for example,
    # if you realize you made a mistake.
    session.cancel()
```

<span id="mult-jobs-in-session"></span>
## Invoke multiple primitives in a session

> A session can handle multiple primitives, allowing for various operations within a single session.  The following example shows how you can create both an instance of the `Sampler` class and one of the `Estimator` class and invoke their `run()` methods within a session.

Reviewer: ```suggestion
A session can handle multiple primitives, allowing for various operations within a single session.  The following example shows how you can create both an instance of the `Sampler` class and one of the `Estimator` class, then invoke their `run()` methods within a session.
```

  </TabItem>
</Tabs>

<span id="two-vqe"></span>
## Run two VQE algorithms in a session by using threading

> 

Reviewer 3: ```suggestion

You can get more out of a session by running multiple workloads at once. The following example shows how you can run two VQE algorithms, each uses a different classical optimizer, simultaneously inside a single session. Job tags are also used to differentiate jobs from each workload. 
```

## Next steps

<Admonition type="tip" title="Recommendations">
  - Try an example in the [Quantum approximate optimization algorithm (QAOA)](https://learning.quantum.ibm.com/tutorial/quantum-approximate-optimization-algorithm) tutorial.

>   - Review the [Session API](https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.Session) reference.

Reviewer: ```suggestion
  - Review the [Session API](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.Session) reference.
```
